optuna/integration/skopt.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import absolute_import
+
+import numpy as np
+
+from optuna import distributions
+from optuna.samplers import BaseSampler
+from optuna.samplers import TPESampler
+from optuna.structs import StudyDirection
+from optuna import types
+
+try:
+    import skopt
+    from skopt.space import space
+
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # SkoptSampler is disabled because Scikit-Optimize is not available.
+    _available = False
+
+if types.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import List  # NOQA
+    from typing import Optional  # NOQA
+    from typing import Set  # NOQA
+    from typing import Tuple  # NOQA
+    from typing import Union  # NOQA
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+    from optuna.study import InTrialStudy  # NOQA
+
+
+class SkoptSampler(BaseSampler):
+    """Sampler using Scikit-Optimize as the backend.
+
+    Example:
+
+        Optimize a simple quadratic function by using :class:`~optuna.integration.SkoptSampler`.
+
+        .. code::
+
+                def objective(trial):
+                    x = trial.suggest_uniform('x', -100, 100)
+                    y = trial.suggest_int('y', -10, 10)
+                    return x**2 + y
+
+                sampler = optuna.integration.SkoptSampler()
+                study = optuna.create_study(sampler=sampler)
+                study.optimize(objective, n_trials=100)
+
+    Args:
+        independent_sampler:
+            A :class:`~optuna.samplers.BaseSampler` instance that is used for independently
+            sampling parameters unknown to :class:`~optuna.integration.SkoptSampler`.
+            An "unknown parameter" means a parameter that isn't contained in
+            :meth:`~optuna.study.InTrialStudy.product_search_space` of the target study.
+
+            If :obj:`None` is specified, :class:`~optuna.samplers.TPESampler` is used
+            as the default. See also :class:`~optuna.samplers`.
+        skopt_kwargs:
+            Keyword arguments passed to the constructor of
+            `skopt.Optimizer <https://scikit-optimize.github.io/#skopt.Optimizer>`_
+            class.
+
+            Note that the ``dimensions`` argument is automatically added by
+            :class:`~optuna.integration.SkoptSampler`, so you don't have to specify it
+            (it will be ignored even if specified).
```

Comment:
I feel that this message can be simpler. For instance, what do you think of the following sentence?

> Note that ``dimensions`` argument in ``skopt_kwargs`` will be ignored because it is added by :class:`~optuna.integration.SkoptSampler` automatically.

Reply:
Seems good, thank you!

Fixed: https://github.com/pfnet/optuna/pull/399/commits/e0ca9ac44bb8dedf38bdcb2d6d73afcb5c99da7e